tactics/autorewrite.ml

```diff
@@ -424,6 +430,25 @@ let gen_auto_multi_rewrite conds tac_main lbas cl =
 let auto_multi_rewrite ?(conds=Naive) lems cl =
   Proofview.wrap_exceptions (fun () -> gen_auto_multi_rewrite conds (Proofview.tclUNIT()) lems cl)
 
+(* Same hack as auto hints: we generate an essentially unique identifier for
+   rewrite hints. *)
+let fresh_key =
+  let id = Summary.ref ~name:"REWHINT-COUNTER" 0 in
+  fun () ->
+    let cur = incr id; !id in
+    let lbl = Id.of_string ("_" ^ string_of_int cur) in
+    let kn = Lib.make_kn lbl in
+    let (mp, _) = KerName.repr kn in
+    (* We embed the full path of the kernel name in the label so that
+       the identifier should be unique. This ensures that including
+       two modules together won't confuse the corresponding labels. *)
+    let lbl = Id.of_string_soft (Printf.sprintf "%s#%i"
+      (ModPath.to_string mp) cur)
+    in
+    KerName.make mp (Label.of_id lbl)
+
+let _ = fresh_key
```

Comment:
???

Reply:
It's too hot around here, I forgot to remove stuff from the previous hackish state.

Reply:
Fixed.